tests/test_stack.py

```diff
@@ -19,3 +22,13 @@ def test_led():
 
     for led in r.demo_board.leds:
         led.state = True
+
+
+def test_robot_lock():
+    """Test that we cannot have more than one Robot object."""
+    r1 = Robot()
+
+    assert r1._lock
```

Comment:
What exactly is this asserting. It would be better to assert that the address bound to is what you were expecting.

Reply:
Fair point. Significantly improved.